network/p2p/dns/resolver_test.go

```diff
@@ -76,33 +75,35 @@ func TestResolver_CacheExpiry(t *testing.T) {
 		metrics.NewNoopCollector(),
 		dnsCache,
 		WithBasicResolver(&basicResolver),
-		WithTTL(1*time.Second)) // cache timeout set to 1 seconds for this test
+		WithTTL(3*time.Second)) // cache timeout set to 3 seconds for this test
 
 	cancelCtx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	ctx, _ := irrecoverable.WithSignaler(cancelCtx)
 	resolver.Start(ctx)
 	unittest.RequireCloseBefore(t, resolver.Ready(), 100*time.Millisecond, "could not start dns resolver on time")
 
-	size := 10 // we have 10 txt and 10 ip lookup test cases
-	times := 5 // each domain is queried for resolution 5 times
+	size := 5  // we have 5 txt and 5 ip lookup test cases
+	times := 3 // each domain is queried for resolution 3 times
 	txtTestCases := testnetwork.TxtLookupFixture(size)
 	ipTestCase := testnetwork.IpLookupFixture(size)
 
 	// each domain gets resolved through underlying resolver twice: once initially, and once after expiry.
 	resolverWG := mockBasicResolverForDomains(t, &basicResolver, ipTestCase, txtTestCases, happyPath, 2)
 
+	// queries (5 + 5) cases * 3 = 100 queries.
 	queryWG := syncThenAsyncQuery(t, times, resolver, txtTestCases, ipTestCase, happyPath)
 	unittest.RequireReturnsBefore(t, queryWG.Wait, 1*time.Second, "could not perform all queries on time")
 
-	time.Sleep(2 * time.Second) // waits enough for cache to get invalidated
+	time.Sleep(4 * time.Second) // waits enough for cache to get invalidated
```

Comment:
was the test failing with the lower number? that's a long time to sleep

Reply:
2nd this comment, if we can get away with a smaller timeout can we use a shorter time?

Reply:
Confirming that the test is passing with reduced timeouts: https://github.com/onflow/flow-go/pull/2670/commits/04ac8c71637161270be21fa8ad86f3e30fc33f09